packages/react/src/checkout/draftOrders/hooks/useDraftOrder.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  getDraftOrder,
+  getDraftOrderError,
+  getRemoveDraftOrderError,
+  getUpdateDraftOrderError,
+  isDraftOrderFetched,
+  isLoadingDraftOrder,
+  isLoadingRemoveDraftOrder,
+  isUpdatingDraftOrder,
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import useDraftOrders from './useDraftOrders.js';
+import type {
+  Config,
+  DraftOrder,
+  DraftOrderData,
+  DraftOrderItem,
+  DraftOrdersQuery,
+  PatchDraftOrderItemData,
+} from '@farfetch/blackout-client';
+import type { UseDraftOrderOptions } from './types/index.js';
+
+const useDraftOrder = (
+  draftOrderId: DraftOrder['id'],
+  query: DraftOrdersQuery,
+  options?: UseDraftOrderOptions,
+) => {
+  const draftOrderIdHookParameter = draftOrderId;
+  const queryHookParameter = query;
+  const { enableAutoFetch = true, fetchConfig } = options || {};
+
+  const {
+    actions: { fetchDraftOrder, remove, removeItem, update, updateItem },
+  } = useDraftOrders(
+    { customerId: '' },
+    { enableAutoFetch: false, fetchConfig },
+  );
+
+  /**
+   * Fetches the draft order. You can override the draft order id to fetch by using
+   * the optional `draftOrderId` parameter. However, the output from
+   * the hook will respect the order id passed to it and not the override.
+   *
+   * @param query - Overrides the query draft order from the hook. If undefined, the `query` passed to the hook will be used instead.
+   * @param config - Custom configurations to send to the client instance (axios). If undefined, the `fetchConfig` passed to the hook will be used instead.
+   * @param draftOrderId  - Overrides the draft order id from the hook. If undefined, the `draftOrderId` passed to the hook will be used instead. Note that the output of the hook will respect the `draftOrderId` parameter from the hook.
+   *
+   * @returns Promise that will resolve when the call to the endpoint finishes.
+   */
+  const fetch = useCallback(
+    (
+      query: DraftOrdersQuery = queryHookParameter,
+      config: Config | undefined = fetchConfig,
+      draftOrderId: DraftOrder['id'] = draftOrderIdHookParameter,
```

Comment:
Hi @anapim 

This is a particular case where config should not be the last one. This is because the draftOrderId, query and config parameters are a "escape hatch" for the developer so he can override the ones passed to the hook. In this case, as the most probable thing the developer would try to override in a request is the config parameter since the state returned by the hook will always be in terms of the passed draftOrderId and query passed to it and not as the parameters to the action. So, in this case to avoid having the user specify "undefined" in all args to reach the config, we change the order here. This is being done in other hooks as well (For example useOrder).

Reply:
I understand. But why is it the second parameter here, between the query and draftOrderId?

Reply:
Basically, here we are exporting actions from useDraftOrders in `useDraftOrder`, if you look at `draftOrderId`, its value comes as a require from the hook itself received as a parameter. The correct thing to do is for the actions to add the require parameters and configs before the parameters received by the hook. 
This point has been adjusted. To understand this better, you can read the JsDocs of the actions in `useDraftOrder`

Reply:
@escabora , the config should come as first as @anapim noted. Can you change this please?

Reply:
This has already been changed